crates/steel-parser/src/parser.rs

```diff
@@ -1120,231 +1090,258 @@ fn wrap_in_doc_function(expr: ExprKind, comment: String) -> ExprKind {
 
 impl<'a> Parser<'a> {
     fn get_next_and_maybe_wrap_in_doc(&mut self) -> Option<Result<ExprKind>> {
-        let mut next;
+        let mut datum_comments = vec![];
 
-        loop {
-            next = self.tokenizer.next();
+        macro_rules! maybe_return {
+            ($value:expr) => {{
+                let value = $value;
 
-            if let Some(res) = next {
-                match res.ty {
-                    TokenType::Comment => {
-                        if self.comment_buffer.is_empty()
-                            && !self.collecting_comments
-                            && res.source().trim_start_matches(';').starts_with("@doc")
-                        {
-                            self.collecting_comments = true;
+                if value.is_ok() {
+                    if datum_comments.len() > 0 {
+                        let _ = datum_comments.pop();
 
-                            continue;
-                        }
+                        continue;
+                    }
+                }
 
-                        if self.collecting_comments {
-                            let doc_line = res.source().trim_start_matches(';');
+                return Some(value);
+            }};
+        }
 
-                            // If we hit another comment, clear it
-                            if doc_line.starts_with("@doc") {
-                                // println!("Clearing buffer");
+        while let Some(res) = self.tokenizer.next() {
```

Comment:
Again, very noisy diff. Most changes are going from `return Some(value)` to `maybe_return![value]`.